dnsaas/admin.py

```diff
@@ -0,0 +1,427 @@
+    CryptoKey,
+    Domain,
+    DomainMetadata,
+    Record,
+    SuperMaster,
+)
+from powerdns.models.tsigkeys import TsigKey
+from powerdns.utils import Owned, DomainForRecordValidator, is_owner
+
+
+admin_site2 = AdminSite('admin2')
+
+
+#TODO: is this really needed?
+class CopyingAdmin(admin.ModelAdmin):
+
+    field_prefix = ''
+    target_prefix = ''
+
+    def get_form(self, request, obj=None, **kwargs):
+        form = super().get_form(request, obj, **kwargs)
+        from_pk = request.GET.get(self.from_field)
+        if from_pk is not None:
+            self.from_object = self.FromModel.objects.get(pk=from_pk)
+            for field in self.CopyFieldsModel.copy_fields:
+                form.base_fields[field[len(self.field_prefix):]].initial = \
+                    getattr(self.from_object, field[len(self.target_prefix):])
+        else:
+            self.from_object = None
+        return form
+
+
+from rules.contrib.admin import ObjectPermissionsModelAdmin
+class OwnedAdmin(ForeignKeyAutocompleteAdmin, ObjectPermissionsModelAdmin):
+    """Admin for models with owner field"""
+
+    def save_model(self, request, object_, form, change):
+        if object_.owner is None:
+            object_.owner = request.user
+        #TODO: rm it?
+        #object_.email_owner(request.user)
+        super(OwnedAdmin, self).save_model(request, object_, form, change)
+
+    #def get_related_filter(self, model, request):
+    #    #TODO:: i assume this fn is up to remove
+    #    return super(OwnedAdmin, self).get_related_filter(model, request)
+    #    user = request.user
+    #    if not issubclass(model, Owned) or rules.is_superuser(user):
+    #        return super(OwnedAdmin, self).get_related_filter(model, request)
+    #    return models.Q(owner=user)
+
+
+
+
+from django.utils.translation import ugettext_lazy as _
+from django.db import models
+try:
+    from django.contrib.admin import SimpleListFilter
+except ImportError:
+    _domain_filters = ('type', 'last_check', 'account',)
+else:
+    class ReverseDomainListFilter(SimpleListFilter):
+        title = _('domain class')
+
+        # Parameter for the filter that will be used in the URL query.
+        parameter_name = 'domain_class'
+
+        def lookups(self, request, model_admin):
+            return (
+                ('fwd', _('domain:forward')),
+                ('rev', _('domain:reverse')),
+            )
+
+        def queryset(self, request, queryset):
+            q = (models.Q(name__endswith='.in-addr.arpa') |
+                 models.Q(name__endswith='.ip6.arpa'))
+            if self.value() == 'fwd':
+                return queryset.exclude(q)
+            if self.value() == 'rev':
+                return queryset.filter(q)
+    _domain_filters = (
+        ReverseDomainListFilter, 'type', 'last_check', 'account',
+    )
+
+
+class DomainMetadataInline(admin.TabularInline):
+    model = DomainMetadata
+    extra = 0
+
+
+class DomainAdmin(OwnedAdmin, CopyingAdmin):
```

Comment:
domain will soon be not-owned, so imo there is no need to mix `OwnedAdmin` and handle owner logic here anymore.